index/scorch/segment/zap/docvalues.go

```diff
@@ -254,20 +275,22 @@ func (s *SegmentBase) VisitDocumentFieldTerms(localDocNum uint64, fields []strin
 
 		if dvIter, exists := s.fieldDvReaders[fieldIDPlus1-1]; exists &&
 			dvIter != nil {
-			dvIterClone = dvIter.cloneInto(dvIterClone)
+			if _, ok := dvs.dvrs[fieldIDPlus1-1]; !ok {
+				dvs.dvrs[fieldIDPlus1-1] = dvIter.cloneInto(dvs.dvrs[fieldIDPlus1-1])
+			}
 
 			// check if the chunk is already loaded
-			if docInChunk != dvIterClone.curChunkNumber() {
-				err := dvIterClone.loadDvChunk(docInChunk, s)
+			if docInChunk != dvs.dvrs[fieldIDPlus1-1].curChunkNumber() {
+				err := dvs.dvrs[fieldIDPlus1-1].loadDvChunk(docInChunk, &s.SegmentBase)
 				if err != nil {
 					continue
```

Comment:
even if the dvReader load fails for a Field, we may attempt to try it for other fields requested is the motive here.

Reply:
ok, that's fine, but what happens to that first field?  will fail to return correct data for it silently?

Reply:
hm..so far I thought that was fine as for some reason my impression was that we are not really propagating an err at the collector level for this. But now when I question myself about this I am not sure whether this is correct..

Reply:
one thought (which i can't confirm because i don't really understand the flow any more) is that it would be OK if we knew what we'd end up un-inverting the field at runtime to mask the problem.  but i don't know if thats the case or not.

Reply:
for non persisted doc value fields, the control won't come here as it is handled at a segment snapshot level cache and this is only for already persisted docValues. And then a read error could be treated like a major and bail out right there..

Reply:
I've taken a look at `loadDvChunk` and it seems the only case it returns an error is when we fail to decode a Uvarint at a location where we expect one.  I think this should be considered fatal as it suggests some sort of corruption.

Reply:
done!